### PR TITLE
concepts: Add GND

### PR DESCRIPTION
**Pull request by rerowep**
* Dependencies update.
    * Deletes "identifier" from agents.
    * Adds GND to concepts.
    * Corrects GND for places
    * Adds IDREF bf:Temporal for 008 Tz5.

### CODE REVIEW

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    """Get identifier and id from (identifier)id."""
+    if match := re_source_and_id.match(data):
+        with contextlib.suppress(IndexError):
+            source = match.group(1).upper().replace("FRPBN", "BNF")
+            return source, match.group(2)
+    return None, None
+
+
+class Transformation(object):
+    """Transformation MARC21 to JSON for GND autority person."""
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:53]: "for GND concept entity"

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    #     - u Werk
+    #     """
+    #     for field_075 in self.marc.get_fields("075") or []:
+    #         if field_075.get("2") and field_075["2"] == "gndgen":
+    #             return RECORD_TYPES.get(field_075["s"])
+
+    def _transform(self):
+        """Call the transformation functions."""
+        record_type = "bf:Topic"
+        if record_type in {"bf:Topic", "bf:Temporal"}:
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:83]: This "if" is useless

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+            for func in dir(self):
+                if func.startswith("trans"):
+                    func = getattr(self, func)
+                    func()
+        else:
+            msg = f"Not a contribution: {record_type}"
+            if self.logger and self.verbose:
+                self.logger.warning(f"NO TRANSFORMATION: {msg}")
+            self.json_dict = {"NO TRANSFORMATION": msg}
+            self.trans_gnd_pid()
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:93]: Delete this

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+                    elif subfield_i == "exakte Aequivalenz":
+                        exacte = {
+                            "source": source,
+                            "type": "bf:Nbn",
+                            "value": id_,
+                        }
+        if exacte:
+            self.json_dict.setdefault("identifiedBy", []).append(exacte)
+        for equivalence in equivalences:
+            self.json_dict.setdefault("identifiedBy", []).append(equivalence)
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:168]: I don't exactly understand how can I differentiate in the JSON data an standard equivalence and an exact equivalence? Also, the data in 750 is and equivalence, not identifiedBy. See below

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+            if subfield_0 and subfield_2:
+                self.json_dict.setdefault("identifiedBy", []).append(
+                    {
+                        "source": subfield_2.upper(),
+                        "type": "uri",
+                        "value": subfield_0,
+                    }
+                )
+        equivalences = []
+        exacte = {}
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:145]: name of the function : `exact`

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+                subfields_0 = field_682.get_subfields("0")
+                for subfield_0 in subfields_0:
+                    if subfield_0.startswith("(DE-101)"):
+                        self.json_dict["relation_pid"] = {
+                            "value": subfield_0.replace("(DE-101)", ""),
+                            "type": "redirect_to",
+                        }
+
+    def trans_gnd_relation(self):
+        """Transformation relation 551."""
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:243]: Shouldn't it be 550?

File: rero_mef/marctojson/do_gnd_concepts.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+            if value:
+                self.json_dict[relation] = value
+
+    def trans_gnd_classification(self):
+        """Transformation classification from field 686."""
+        if self.logger and self.verbose:
+            self.logger.info("Call Function", "trans_gnd_classification")
+        # TODO: find classification
+
+    def trans_gnd_close_match(self):
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_gnd_concepts.py:286]: You already put the 750 in `identifiedBy`, we should decide if we put it there or in "closeMatch/exactMatch"

**Review by PascalRepond — CHANGES_REQUESTED** [review]

File: rero_mef/marctojson/do_gnd_concepts.py (hunk 4 above)
Comment (rerowep) [reply rero_mef/marctojson/do_gnd_concepts.py]: Added `exactMatch`.

File: rero_mef/places/mef/mappings/v7/places_mef/mef-place-v0.0.1.json [hunk 8]
```diff
@@ -119,6 +147,9 @@
                   },
                   "value": {
                     "type": "keyword"
+                  },
+                  "identifier": {
```
Comment (PascalRepond) [inline rero_mef/places/mef/mappings/v7/places_mef/mef-place-v0.0.1.json:151]: How about calling this field `_text`? This way we know that it is constructed and specific to MEF and not to be confused with `identifiedBy.value`.

**Review by PascalRepond — APPROVED** [review]

File: rero_mef/concepts/api.py [hunk 9]
```diff
@@ -63,16 +66,72 @@ def create_or_update_mef(self, dbcommit=False, reindex=False):
         :param reindex: Reindex record.
         :returns: MEF record, MEF action
         """
-        from rero_mef.concepts import ConceptMefRecord
+        actions = {}
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_mef/concepts/api.py]: circular import

**Pull request merged**